src/Giraffe/Core.fs

```diff
@@ -435,8 +435,7 @@ module Core =
     /// <param name="str">The string value to be send back to the client.</param>
     /// <returns>A Giraffe <see cref="HttpHandler" /> function which can be composed into a bigger web application.</returns>
     let setBodyFromString (str: string) : HttpHandler =
-        let bytes = Encoding.UTF8.GetBytes str
-        fun (_: HttpFunc) (ctx: HttpContext) -> ctx.WriteBytesAsync bytes
+        fun (_: HttpFunc) (ctx: HttpContext) -> ctx.WriteStringAsync str
```

Comment:
Indeed, this is equivalent since `WriteStringAsync` implementation is:

```fsharp
/// <summary>
/// Writes an UTF-8 encoded string to the body of the HTTP response and sets the HTTP Content-Length header accordingly.
/// </summary>
/// <param name="ctx">The current http context object.</param>
/// <param name="str">The string value to be send back to the client.</param>
/// <returns>Task of Some HttpContext after writing to the body of the response.</returns>
[<Extension>]
static member WriteStringAsync(ctx: HttpContext, str: string) =
    ctx.WriteBytesAsync(Encoding.UTF8.GetBytes str)
```

Good catch.